components/movementsensor/replay/replay.go

```diff
@@ -567,7 +569,9 @@ func (replay *replayMovementSensor) attemptToGetData(ctx context.Context, method
 	if replay.closed {
 		return false, errSessionClosed
 	}
-	if err := replay.updateCache(ctx, method); err != nil && !strings.Contains(err.Error(), ErrEndOfDataset.Error()) {
+	cancelCtx, cancel := context.WithTimeout(context.Background(), tabularDataByFilterTimeout)
+	defer cancel()
```

Comment:
Moved the new `cancelCtx` creation into `attemptToGetData` since we want to create this new timeout context for each call to `updateCache` in here.